blob/local_store.go

```diff
@@ -94,3 +96,47 @@ func (l *LocalStore) Describe(ctx context.Context, id ID) (*Descriptor, error) {
 		ModificationTime: stat.ModTime(),
 	}, nil
 }
+
+// Lists all locally stored blob IDs. Files in the local store directory which
+// do not end in ".bin", or which cannot be parsed into a blob ID, will not be
+// included.
+func (l *LocalStore) List(ctx context.Context) ([]ID, error) {
+	var ids []ID
+
+	dir, err := os.ReadDir(l.dir)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read local store directory: %w", err)
+	}
+
+	for _, binFileEntry := range dir {
+		idString, isBin := strings.CutSuffix(binFileEntry.Name(), ".bin")
```

Comment:
Add a check for `IsDir()` to skip directories named with `.bin` (not that there ever should be):
```suggestion
		if binFileEntry.IsDir() {
			continue
		}
		idString, isBin := strings.CutSuffix(binFileEntry.Name(), ".bin")
```